engine/clients/qdrant/configure.py

```diff
@@ -65,8 +88,20 @@ def recreate(self, dataset: Dataset, collection_params):
             ),
         )
         for field_name, field_type in dataset.config.schema.items():
-            self.client.create_payload_index(
-                collection_name=QDRANT_COLLECTION_NAME,
-                field_name=field_name,
-                field_schema=self.INDEX_TYPE_MAPPING.get(field_type),
-            )
+            is_tenant = True if field_name in dataset.config.tenants else False
+            if field_type in ["keyword", "uuid"]:
+                self.client.create_payload_index(
+                    collection_name=QDRANT_COLLECTION_NAME,
+                    field_name=field_name,
+                    field_schema=self.INDEX_PARAMS_TYPE_MAPPING.get(field_type)(
+                        type=self.INDEX_TYPE_MAPPING.get(field_type),
+                        is_tenant=is_tenant,
+                        on_disk=self.payload_index_params.get("on_disk", False),
+                    ),
+                )
+            else:
+                self.client.create_payload_index(
+                    collection_name=QDRANT_COLLECTION_NAME,
+                    field_name=field_name,
+                    field_schema=self.INDEX_TYPE_MAPPING.get(field_type),
+                )
```

Comment:
Why not allow `on_disk` for other fields?

Reply:
I think we need granular control in the long run. But I'm okay with approving this for now.

Reply:
Right now not all types of payload indexes allow on_disk. So in this PR I only allow "keyword" and "uuid" to have that. When we have experiments for other types we can alter this place.